docs/getting-started/first-harper-app.md

```diff
@@ -0,0 +1,182 @@
+      } catch (err) {
+        return { error: "Invalid JSON" };
+      }
+    }
+    
+    // Validate required fields
+    if (!data.title || !data.author) {
+      return { error: "Title and author are required" };
+    }
+    
+    // Prevent primary key overriding
+    if (data.id) delete data.id;
+    
+    try {
+      const result = await this.table.post(data);
+      return result;
+    } catch (error) {
+      return { error: "Error creating book", details: error.message };
+    }
+  }
+}
+```
+Let's break down what this JavaScript file is doing:
+
+### Resource Class Extension
+```js
+export class Books extends Resource {
+```
+
+This line creates a custom `Books` class that extends Harper's built-in `Resource` class. By doing this, we inherit all the standard functionality while allowing us to customize behavior.
+
+### GET Method
+```js
+async get() {
+  console.log("GET request received");
+  const id = this.getId();
+  
+  if (id) {
+    // Return a single book
+    return await this.table.get(id);
+  } else {
+    // Return all books with optional genre filter
+    const genre = this.getQueryParam("genre");
+    const searchOptions = genre ? { genre } : {};
+    return await this.table.search(searchOptions);
+  }
+}
+```
+
+The `get()` method handles HTTP GET requests to our Book endpoint. It:
+- Logs that a request was received (helpful for debugging)
+- Checks if an ID was provided in the URL path
+- If an ID exists, returns a single book by that ID
+- If no ID exists, checks for a `genre` query parameter
+- Returns all books, filtered by genre if specified
+
+### POST Method
+```js
+async post(data) {
+  console.log("POST request received");
+  
+  // Parse string data if needed
+  if (typeof data === "string") {
+    try {
+      data = JSON.parse(data);
+    } catch (err) {
+      return { error: "Invalid JSON" };
+    }
+  }
+  
+  // Validate required fields
+  if (!data.title || !data.author) {
+    return { error: "Title and author are required" };
+  }
+  
+  // Prevent primary key overriding
+  if (data.id) delete data.id;
+  
+  try {
+    const result = await this.table.post(data);
+    return result;
+  } catch (error) {
+    return { error: "Error creating book", details: error.message };
+  }
+}
+```
+
+The `post()` method handles HTTP POST requests to create new books. It:
+- Handles both string and object data formats
+- Validates that required fields (`title` and `author`) are present
+- Removes any provided `id` field to prevent primary key conflicts
+- Creates the book record in the database
+- Returns the result or an error message if something goes wrong
+
+## Run and Test
+1. Start your application:
+    ```bash
+    harperdb dev .
+    ```
+2. Create a book:
+    ```bash\
+    curl -X POST -H "Content-Type: application/json" \ -d '{"title": "The Great Gatsby", "author": "F. Scott Fitzgerald", "publishedYear": 1925, "genre": "Fiction"}' \ http://localhost:9926/Books
+    ```
```

Comment:
```suggestion
    ```bash
    curl http://localhost:9926/Books \
      -X POST \
      -H "Content-Type: application/json" \
      -d '{"title": "The Great Gatsby", "author": "F. Scott Fitzgerald", "publishedYear": 1925, "genre": "Fiction"}'
    ```
```

I prefer spreading these out multiline so they don't deal with word-wrap or overflow